src/App.js

```diff
@@ -14,11 +16,87 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+export class App extends Component {
+  state = {
+    goods: [...goodsFromServer],
+    visible: false,
+  };
+
+  startApp = () => {
+    this.setState({ visible: true });
+  };
+
+  reverseList = () => {
+    this.setState(state => ({
+      goods: [...state.goods].reverse(),
```

Comment:
Let's move all the sortings inside the `GoodsList` component.
It's good to split the logic. `App` component should only have the button 'start' which will toggle the render of the `GoodsList`. There you should pass initial goods and sort them inside that component (not inside the `App`)